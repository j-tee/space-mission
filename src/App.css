#root {
  margin: 0 3rem 0 3rem;
}

.rocket-card {
  display: flex;
  flex-direction: row;
  gap: 2rem;
  margin-bottom: 2rem;
}

.rocket-desc {
  display: flex;
  flex-direction: column;
}

.miision-btn {
  width: 10rem;
}

.join-mission-btn {
  color: #585c61;
  border: 2px solid #585c61;
}

.wrapper {
  display: flex;
  justify-content: space-around;
}

.wrapper table tr {
  border: 2px solid #dee2e6;
}

.leave-mission-btn {
  color: #e7757f;
  border: 2px solid #e7757f;
}

.align-btn {
  width: 10rem;
  vertical-align: middle;
}

.align-badge {
  vertical-align: middle;
}

.badge {
  background-color: #6e767e;
  margin: 0;
}

.membership {
  display: block;
}

.rocket-reserve {
  background-color: #18a2b8;
  margin-right: 1rem;
  padding: 0.1rem;
  border-radius: 0.2rem;
  color: white;
  font-size: 0.8rem;
}

.cancel-btn {
  background-color: #fff;
  border: 2px solid red;
  color: #92a0ad;
}

.btn {
  cursor: pointer;
  padding: 8px;
  border-radius: 3px;
  font-size: 0.813rem;
  font-weight: 700;
  width: 10rem;
}

.reserve-btn {
  background-color: #0290ff;
  color: #fff;
  border: none;
}

.image {
  width: 12rem;
  height: 9rem;
  object-fit: cover;
}

.header {
  display: flex;
  justify-content: space-between;
}

.banner {
  font-weight: 700;
  font-size: 1.3rem;
}

.info {
  margin-bottom: 5px;
}

.title-desc {
  font-size: 1.5rem;
  margin-bottom: 0;
}

.nav {
  padding: 1rem;
}

.logo-title {
  display: flex;
  justify-content: space-between;
  gap: 2rem;
  padding: 1rem 0 1rem 0;
}

.logo-title span img {
  height: 2rem;
  width: 2rem;
}

ul {
  display: flex;
  flex-direction: row;
  gap: 2rem;
  list-style: none;
}

ul li:last-child {
  border-left: 2px solid black;
  padding-left: 2rem;
}

.cancel-mission {
  border-style: solid;
  border-width: 1px;
  border-color: #6e767e;
  border-radius: 4px;
  background: #f3fef3;
  padding: 0.5rem;
  color: red;
  font-size: 1rem;
}
