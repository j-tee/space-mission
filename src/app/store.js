/* eslint-disable */

import { configureStore } from '@reduxjs/toolkit';

export  const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});
