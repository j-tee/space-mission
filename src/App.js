/* eslint-disable */
import React from 'react';
import { Route, Routes } from 'react-router';
import './App.css';
import Missions from './components/Missions/Missions';
import Rockets from './components/Rocket/Rockets';
import MyProfile from './components/MyProfile';
import Navbar from './components/navigation/Navbar';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <Navbar />
        <Routes>
          <Route path="/" element={<Rockets />} />
          <Route path="/Missions" element={<Missions />} />
          <Route path="/MyProfile" element={<MyProfile />} />
        </Routes>
      </header>
    </div>
  );
}

export default App;
