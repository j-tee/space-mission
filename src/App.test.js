import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import store from './redux/configureStore';

test('renders App component correctly', () => {
  const Component = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );

  expect(Component).toMatchSnapshot();
});
