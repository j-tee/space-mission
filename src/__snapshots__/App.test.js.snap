// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <div
            class="header"
          >
            <div
              class="logo-title"
            >
              <span>
                <img
                  alt="logo"
                  src="planet.png"
                />
              </span>
              <span
                class="banner"
              >
                Space Traveller's Hub
              </span>
            </div>
            <div
              class="nav"
            >
              <ul
                class="nav-bar"
              >
                <li>
                  <a
                    activeclassname="active-link"
                    aria-current="page"
                    class="active"
                    href="/"
                  >
                    Rockets
                  </a>
                </li>
                <li>
                  <a
                    activeclassname="active-link"
                    class=""
                    href="/missions"
                  >
                    Missions
                  </a>
                </li>
                <li>
                  <a
                    activeclassname="active-link"
                    class=""
                    href="/MyProfile"
                  >
                    My Profile
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <div
          class="header"
        >
          <div
            class="logo-title"
          >
            <span>
              <img
                alt="logo"
                src="planet.png"
              />
            </span>
            <span
              class="banner"
            >
              Space Traveller's Hub
            </span>
          </div>
          <div
            class="nav"
          >
            <ul
              class="nav-bar"
            >
              <li>
                <a
                  activeclassname="active-link"
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  Rockets
                </a>
              </li>
              <li>
                <a
                  activeclassname="active-link"
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  activeclassname="active-link"
                  class=""
                  href="/MyProfile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
